### PR TITLE
Added status_updates_enabled attribute to twingate_connector 

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/291

## Changes
- added new optional attribute `status_updates_enabled`

### CODE REVIEW

File: twingate/internal/client/connector.go [hunk 1]
```diff
@@ -10,57 +10,92 @@ import (
...
 
-func (client *Client) CreateConnector(ctx context.Context, remoteNetworkID, connectorName string) (*model.Connector, error) {
-	if remoteNetworkID == "" {
+func (client *Client) CreateConnector(ctx context.Context, input *model.Connector) (*model.Connector, error) {
+	if input == nil || input.NetworkID == "" {
 		return nil, NewAPIError(ErrGraphqlNetworkIDIsEmpty, "create", connectorResourceName)
 	}
 
+	if input.StatusUpdatesEnabled != nil {
+		return client.createConnectorWithNotificationStatus(ctx, input)
```
Comment (ekampf) [inline twingate/internal/client/connector.go:19]: This doesn't seem like the right solution...  what if we had more optional variables?   we'll need to create a query struct and function for every possible permutation?

File: twingate/internal/client/query/connector-create.go [hunk 2]
```diff
@@ -4,6 +4,10 @@ type CreateConnector struct {
 	ConnectorEntityResponse `graphql:"connectorCreate(remoteNetworkId: $remoteNetworkId, name: $connectorName)"`
 }
 
+type CreateConnectorWithNotificationStatus struct {
+	ConnectorEntityResponse `graphql:"connectorCreate(remoteNetworkId: $remoteNetworkId, name: $connectorName, hasStatusNotificationsEnabled: $hasStatusNotificationsEnabled)"`
```
Comment (ekampf) [inline twingate/internal/client/query/connector-create.go:8]: does `go-graphql` support defining [defaults](https://graphql.org/learn/queries/#default-variables) here?
```
graphql:"connectorCreate(remoteNetworkId: $remoteNetworkId, name: $connectorName, hasStatusNotificationsEnabled: $hasStatusNotificationsEnabled = true)"
```

Comment (vmanilo) [reply twingate/internal/client/query/connector-create.go:8]: no, it's not supported

File: twingate/internal/client/connector.go (hunk 1 above)
Comment (vmanilo) [reply twingate/internal/client/connector.go:19]: we can set default value like this, 
but the downside that if you change default value on the backend side, we would need to update client as well

```
        statusUpdatesEnabled := true	// default value
	if input.StatusUpdatesEnabled != nil {
		statusUpdatesEnabled = *input.StatusUpdatesEnabled
	}

	variables := newVars(
		gqlID(input.NetworkID, "remoteNetworkId"),
		gqlNullable(input.Name, "connectorName"),
		gqlVar(statusUpdatesEnabled, "hasStatusNotificationsEnabled"),
	)
```

Comment (vmanilo) [reply twingate/internal/client/connector.go:19]: in the issue there is one point listed

```
If the argument is not specified, the default value must match the API behavior 

(which is true in this case, but in principle could be changed at the API level in the future).
```

Comment (vmanilo) [reply twingate/internal/client/connector.go]: @ekampf it's fixed now, thank you for the backend support :)

**Review by ekampf — APPROVED** [review]

**Pull request merged**